core/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -223,19 +224,21 @@ private void ensureNoPre019State() throws Exception {
      * index metadata with new settings.
      */
     private void pre20Upgrade() throws Exception {
-        MetaData metaData = loadMetaState();
-        List<IndexMetaData> updateIndexMetaData = new ArrayList<>();
-        for (IndexMetaData indexMetaData : metaData) {
+        List<PersistedIndexMetaData> metaData = metaStateService.loadIndicesStatesNoFolderNameValidation();
```

Comment:
why do we need the new loadIndicesStatesNoFolderNameValidation ? what was wrong with loadMetaState() exactly?